docs/plugins.rst

```diff
@@ -1,260 +1,133 @@
 
-- A function decorated with ``respond_to`` is called when a message matching the pattern is sent to the bot (direct message or @botname in a channel/group chat)
-- A function decorated with ``listen_to`` is called when a message matching the pattern is sent on a channel/group chat (not directly sent to the bot)
-- A function decorated with ``at_start`` is called as soon as the plugin is initialized (when the bot starts)
+        from mmpy_bot.plugins.base import Plugin, listen_to
+        from mmpy_bot.wrappers import Message
 
-.. code-block:: python
 
-    import re
+#. Next create a class with the name of your plugin and create an `asyncio` method with the `@listen_to` decorator:
 
-    from mmpy_bot.bot import listen_to
-    from mmpy_bot.bot import respond_to
+    .. code-block:: python
 
+        class MyPlugin(Plugin):
 
-    @respond_to('hi', re.IGNORECASE)
-    def hi(message):
-        message.reply('I can understand hi or HI!')
+            @listen_to("wake up")
+            async def wake_up(self, message: Message):
+                self.driver.reply_to(message, "I'm awake!")
 
+    In the above code block, the `@listen_to` decorator tells the bot to listen on any channel for the string "wake up", and respond with "I'm awake!".
 
-    @respond_to('I love you')
-    def love(message):
-        message.reply('I love you too!')
+#. Now open up `./plugins/__init__.py` and add your plugin module as follows:
 
+    .. code-block:: python
 
-    @listen_to('Can someone help me?')
-    def help_me(message):
-        # Message is replied to the sender (prefixed with @user)
-        message.reply('Yes, I can!')
+        from mmpy_bot.plugins.base import Plugin
+        from mmpy_bot.plugins.example import ExamplePlugin
+        from mmpy_bot.plugins.webhook_example import WebHookExample
+        from mmpy_bot.plugins.my_plugin import MyPlugin
 
-        # Message is sent on the channel
-        # message.send('I can help everybody!')
+        __all__ = ["Plugin", "ExamplePlugin", "WebHookExample", "MyPlugin"]
 
-    @at_start
-    def hello(client):
-        # Note that contrary to respond_to and listen_to @at_start
-        # receives a client object and not a message object
-        team = client.api.get_team_by_name("TESTTEAM")
-        channel = client.api.get_channel_by_name("bot_test", team["id"])
-        client.channel_msg(channel["id"], "Hello, feels good to be alive!!")
+#. Last but not least, you will need to ensure your plugin is started by the bot. In this example we will update the default `bot.py` file
+   to import `MyPlugin`:
 
-To extract params from the message, you can use regular expression:
+    .. code-block:: python
 
-.. code-block:: python
+        from mmpy_bot.plugins import ExamplePlugin, Plugin, WebHookExample, MyPlugin
 
-    from mmpy_bot.bot import respond_to
+        def __init__(self, settings=Settings(), plugins=[ExamplePlugin(), WebHookExample(),
+                     MyPlugin()]
```

Comment:
As mentioned above, they don't need to modify any files other than their main file. They can just have a file like this:

```python
from mmpy_bot import Bot, Settings
from my_plugin import MyPlugin

bot = Bot(settings=Settings(BOT_TOKEN="my_token"), plugins=[MyPlugin()])
bot.run()
```

Reply:
Thanks, that makes a lot more sense!

Reply:
I've created an `entrypoint.py` in #162 that we can refer to in the docs